rules-emerging-threats/2024/Exploits/CVE-2024-3094/proc_creation_lnx_exploit_cve_2024_3094_sshd_child_process.yml

```diff
@@ -0,0 +1,30 @@
+title: Potential Exploitation of CVE-2024-3094 - Suspicious SSH Child Process
+id: 9aa27839-e8ba-4d7a-ac1a-746c22c3d1e5
+status: experimental
+description: |
+    Detects potentially suspicious child process of SSH process (sshd) with a specific execution user. This could be a sign of potential exploitation of CVE-2024-3094.
+references:
+    - https://github.com/amlweems/xzbot?tab=readme-ov-file#backdoor-demo
+author: Arnim Rupp
+date: 2024/04/01
+tags:
+    - attack.execution
+    - cve.2024.3094
+logsource:
+    category: process_creation
+    product: linux
+detection:
+    selection_1:
+        ParentImage|endswith: '/sshd'
+        CommandLine|startswith:
+            - 'bash -c'
+            - 'sh -c'
+        User: 'root'
+    selection_2:
+        ParentImage|endswith: '/sshd'
+        Image|endswith: '/sshd'
+        User: 'sshd'
```

Comment:
@phantinuss can you please double check this selection. See internal discussion for context

Reply:
It's a solid detection with no noise in the environment I have access to.

Is it possible to replace `User: sshd` with a negation of `UID: 0`? Background is that e.g. the CrowdStrike data model has no user name but just the UID field. Furthermore, I observe `UID=0` on almost all Linux hosts I can see, except some that can be baselined easily.

Reply:
on linux mint the uid of sshd is 129.

for selection_2 we could probably add
```
CommandLine|startswith:
                  'sshd: root'
```
if I get this right: https://security.stackexchange.com/questions/115896/can-someone-explain-how-sshd-does-privilege-separation
sshd is the privilege separation user and that process always contains the name of the destination user, in our case root.

Reply:
I don't think that sshd is using a fixed UID. Sampled 7 different distros I have lying around as VMs and they all got different IDs. So it's probably chosen at package install time.

Reply:
Funnily installing on a Ubuntu 22.04 LTS has chosen 129 right now.

Reply:
yep, it's ascending:
```
$ grep :x:12 /etc/passwd
pulse:x:120:128:PulseAudio daemon,,,:/run/pulse:/usr/sbin/nologin
_flatpak:x:121:131:Flatpak system-wide installation helper,,,:/nonexistent:/usr/sbin/nologin
avahi:x:122:132:Avahi mDNS daemon,,,:/run/avahi-daemon:/usr/sbin/nologin
saned:x:123:133::/var/lib/saned:/usr/sbin/nologin
colord:x:124:134:colord colour management daemon,,,:/var/lib/colord:/usr/sbin/nologin
fwupd-refresh:x:125:135:fwupd-refresh user,,,:/run/systemd:/usr/sbin/nologin
hplip:x:126:7:HPLIP system user,,,:/run/hplip:/bin/false
sssd:x:127:137:SSSD system user,,,:/var/lib/sss:/usr/sbin/nologin
xrdp:x:128:140::/run/xrdp:/usr/sbin/nologin
sshd:x:129:65534::/run/sshd:/usr/sbin/nologin
```

Reply:
Just checked the deb postinst code and it's just alling `adduser --system`, so there is no fixed UID we can expect.

Reply:
If you don't match/map the user name you just don't match the rule. Don't think there is the need for splitting. I'll resolve the convo for now and will proceed with merging as this is about a critical issue. We can fine tune it at a later point anyway.